sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -200,12 +198,14 @@ class AstBuilder extends DataTypeAstBuilder
             el.multipartIdentifier().getText.toLowerCase(Locale.ROOT) =>
         withOrigin(bl) {
           throw SqlScriptingErrors.labelsMismatch(
-            CurrentOrigin.get, bl.multipartIdentifier().getText, el.multipartIdentifier().getText)
+            CurrentOrigin.get,
+            toSQLId(bl.multipartIdentifier().getText),
```

Comment:
Just in case, `toSQLId` parses the input string back. It would be better to pass `Seq[String]`, and perform quoting inside of `SqlScriptingErrors`. No need to spread the error specific code here.

Reply:
Changed this for all scripting errors